### PR TITLE
Fix main generator detection pattern

### PR DESCRIPTION
**Pull request by mlenkeit**
This PR contains a fix for #444.

I first covered the existing coding with unit tests, then added the failing test case and the corresponding fix. I also added a new helper function that creates a fake for the `read-pkg-up` module that I use in the tests.

What do you think?

### CODE REVIEW

File: test/router.js [hunk 1]
```diff
@@ -2,11 +2,18 @@
 var assert = require('assert');
 var _ = require('lodash');
 var sinon = require('sinon');
-var Router = require('../lib/router');
+var helpers = require('./helpers');
+var fakeReadPkgUp = helpers.fakeReadPkgUp();
+var proxyquire = require('proxyquire');
+var Router = proxyquire('../lib/router', {
+  'read-pkg-up': fakeReadPkgUp
```
Comment (SBoudrias) [inline test/router.js]: Let's move the mock inline as it is not reusable - or reusable.

**Comment by SBoudrias** [comment]
Thanks for sending this patch!

File: test/router.js (hunk 1 above)
Comment (mlenkeit) [inline test/router.js]: I'm not sure if I understand. Are you suggesting (1) to get ride of line 6 by inlining it in line 9 or (2) to not include the fake read-pkg-up module in helpers but inlining that instead?

Comment (SBoudrias) [inline test/router.js]: I meant inline the `fakeReadPkgUp` function.

**Comment by mlenkeit** [comment]
I inlined the fake read-pkg-up module with 3485f8b21f751709a56e416279130b4548932689 and reverted the commit that introduced the fake to the helpers class with 9f05f4802a540b5f812a5f0e88096d66199fceaa.
Let me know if there's anything else.

**Comment by SBoudrias** [comment]
LGTM, thanks for sending this out 🎉

**Pull request merged**